lib/save.js

```diff
@@ -28,21 +29,35 @@ function save (url, opts) {
     if (err) throw err.message
 
     var pageContent = getPageContent(options.template, page)
-    var dirOutput = path.join(options.output, page.basename)
+    if (options.root) {
+      if (!options.output) {
+        throw new Error("options.root requires options.output")
+      }
+      var dirOutput = path.join(options.output)
+    } else {
+      var dirOutput = path.join(options.output, page.basename)
+    }
 
     var outputContent = createHTML({
       head: '<style>' + options.style + '</style>',
       title: page.title,
       body: pageContent
     })
 
-    try {
-      fs.mkdirSync(dirOutput)
-      fs.writeFileSync(path.join(dirOutput, 'index.html'), outputContent)
-      fs.writeFileSync(path.join(dirOutput, 'index.json'), JSON.stringify(page, { }, 2))
-    } catch (err) {
-      throw new Error('page has already been saved')
-    }
+    return new Promise((resolve, reject) => {
+      mkdirp(dirOutput, (err) => {
+        if (err) { console.error(err); return reject(err) }
+        resolve()
+      })
+    }).then(() => {
+      try {
+        fs.writeFileSync(path.join(dirOutput, 'index.html'), outputContent)
+        fs.writeFileSync(path.join(dirOutput, 'index.json'), JSON.stringify(page, { }, 2))
+      } catch (err) {
+        console.error(err)
```

Comment:
i added this log as i was getting errors that had nothing to do with the page being saved! might be better to just throw the error that is caught rather than creating a new one